test/unit/kubernetes-deploy/kubernetes_resource/pod_test.rb

```diff
@@ -155,6 +155,21 @@ def test_deploy_failed_is_true_for_container_cannot_run_error
     assert_equal expected_msg, pod.failure_message
   end
 
+  def test_deploy_failed_is_false_for_evicted_
+    container_state = {
```

Comment:
Did this test really fail without your change? This is setting container status info, but eviction data (as your business logic correctly assumes) is at the top level of the pod status, like this:

```
status:
  message: 'The node was low on resource: nodefs.'
  phase: Failed
  reason: Evicted
  startTime: 2018-05-04T21:17:08Z
```